src/Transport/Topology/Topologies/TypeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using Settings;
+
+    static class TypeExtensions
+    {
+        // potentially we want to remove this when we simplify hooking customer owned implementations into the topology
+        public static TReturnedType CreateInstance<TReturnedType>(this Type typeToResolve, ReadOnlySettings settings)
```

Comment:
This keeps backwards compatibility:

Following assumptions. Right now with the transport parts container it is possible to:

- Resolve dependencies that are in the parts container (all of them are internal)
- Resolve ReadonlySettings
- Have empty default ctor

From a customer perspective it is only possible to do the following:

- have a ctor on their implementations which required ReadonlySettings or
- have an empty parameterless ctor

Therefore this extension methods keeps this by using either default ctor and calling or assume ctor that has a dependency to readonly settings. In all other cases it blows up